### ISSUE TITLE
GPS Traces options

### ISSUE THREAD

**Issue by MikeParfitt** [MarcusWolschon/osmeditor4android#980]
When requesting the display of GPS traces for the current window, I would like two options
 1) to only download/display GPS traces uploaded by me
 2) the current option to download/display all GPS traces

My suggested wording for these options would be
 1) OSM My GPS traces
 2) OSM All GPS traces

**Comment by simonpoole** [comment]
We currently don't support the downloading of GPS traces at all. Mainly because it is very very slow, as can be seen with JOSM. Not ruling out that it could be supported, but it needs some thought (the GPX layer is a pre-rendered imagery tiles layer).

**Comment by MikeParfitt** [comment]
Fair enough, I hadn't realised that the GPX layer is a set of pre-rendered tiles.

Extracting just the traces uploaded by the relevant user is clearly possible, but extracting only those which intersect with the currently downloaded window might not be so easy.

Plus you have to interact with the OSM database to download each trace, so I now appreciate how slow it could be to show a sub-set of the information that is currently displayed.
________________________________

**Comment by simonpoole** [comment]
I had a look at the current OSM GPX API, essentially there is no way to efficiently download the users GPX traces for a bounding box currently. 

While we can get a list of traces for the logged in user, the results only contain the coordinates for one point (I'm fairly sure it is the 1st one), implying that we would have to download all the traces for which the 1st coordinate is not in the bounding box and explicitly check them for intersection. 

The alternative would be to use the API call that downloads trackpoints, which is not only very slow, it can't be restricted just to the data for the current user. 

In summary, supporting this functionality would need some reworking of the relevant OSM API code and how the GPX data is stored, not all too much work, but IMHO it would only make sense if there is more support in the community for it. Opening https://github.com/openstreetmap/openstreetmap-website/issues/2395 for more input.  

Ref: https://wiki.openstreetmap.org/wiki/API_v0.6#GPS_traces

**Comment by MikeParfitt** [comment]
Hi Simon,

Thanks for following through on this, I have added my reasoning to the
GitHub request.  I think that ultimately, the growing number of GPS tracks on OSM will justify extra filters on :-

GET /api/0.6/trackpoints
GET /api/0.6/user/gpx_files

Perhaps the least contentious might be a date range for when the GPX file was uploaded.

Regards,

Mike
________________________________

**Comment by simonpoole** [comment]
Selective downloading of the users GPX tracks has been implemented in https://github.com/MarcusWolschon/osmeditor4android/commit/4a577b54440af9268c569ae79c9da6731ec5555c
